modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/search/EntriesChecker.java

```diff
@@ -56,22 +62,40 @@ public String getAllRowsCheckBox(HttpServletRequest httpServletRequest) {
 		return null;
 	}
 
+	@Override
+	public Map<String, Object> getData(Object object) {
+		Map<String, Object> data = super.getData(object);
+
+		if (data == null) {
+			return Collections.singletonMap("modelClassName", _getName(object));
+		}
+
+		return HashMapBuilder.create(
+			data
+		).put(
+			"modelClassName", _getName(object)
+		).build();
+	}
```

Comment:
A question about this. Are you using `HashMapBuilder.create(data).put().build()` instead of `data.put()` as a general safe way of working or just because in this specific case you are creating an immutable singleton map to initialize the map? If we are in the second case wouldn't be better to initialize the map with `HashMapBuilder.put().build()`?

Reply:
I'm avoiding the mutation of `data` because the default implementation of `getData` returns the same (mutable) map for different calls on the same instance (and instances are shared). Ideally, no public API should return mutable data (unless it is really necessary), but it is what it is ...

Reply:
Thanks for the explanation